iroh/src/endpoint/quic.rs

```diff
@@ -0,0 +1,445 @@
+        &mut self,
+        factory: Arc<dyn ControllerFactory + Send + Sync + 'static>,
+    ) -> &mut Self {
+        self.0.congestion_controller_factory(factory);
+        self
+    }
+
+    /// Whether to use "Generic Segmentation Offload" to accelerate transmits, when supported by the
+    /// environment
+    ///
+    /// Defaults to `true`.
+    ///
+    /// GSO dramatically reduces CPU consumption when sending large numbers of packets with the same
+    /// headers, such as when transmitting bulk data on a connection. However, it is not supported
+    /// by all network interface drivers or packet inspection tools. `quinn-udp` will attempt to
+    /// disable GSO automatically when unavailable, but this can lead to spurious packet loss at
+    /// startup, temporarily degrading performance.
+    pub fn enable_segmentation_offload(&mut self, enabled: bool) -> &mut Self {
+        self.0.enable_segmentation_offload(enabled);
+        self
+    }
+
+    /// Whether to send observed address reports to peers.
+    ///
+    /// This will aid peers in inferring their reachable address, which in most NATd networks
+    /// will not be easily available to them.
+    pub fn send_observed_address_reports(&mut self, enabled: bool) -> &mut Self {
+        self.0.send_observed_address_reports(enabled);
+        self
+    }
+
+    /// Whether to receive observed address reports from other peers.
+    ///
+    /// Peers with the address discovery extension enabled that are willing to provide observed
+    /// address reports will do so if this transport parameter is set. In general, observed address
+    /// reports cannot be trusted. This, however, can aid the current endpoint in inferring its
+    /// reachable address, which in most NATd networks will not be easily available.
+    pub fn receive_observed_address_reports(&mut self, enabled: bool) -> &mut Self {
+        self.0.receive_observed_address_reports(enabled);
+        self
+    }
+
+    /// Enables the Multipath Extension for QUIC.
+    ///
+    /// Setting this to any nonzero value will enable the Multipath Extension for QUIC,
+    /// <https://datatracker.ietf.org/doc/draft-ietf-quic-multipath/>.
+    ///
+    /// The value provided specifies the number maximum number of paths this endpoint may open
+    /// concurrently when multipath is negotiated. For any path to be opened, the remote must
+    /// enable multipath as well.
+    ///
+    /// Note: setting this value below 13 is not recommended and will currently log a warning.
+    pub fn max_concurrent_multipath_paths(&mut self, max_concurrent: u32) -> &mut Self {
+        if max_concurrent < MAX_MULTIPATH_PATHS + 1 {
+            warn!(
+                "setting QuicTransportConfig::max_concurrent_multipath_paths to {}, below the recommended minimum {}",
+                max_concurrent,
+                MAX_MULTIPATH_PATHS + 1
+            );
+        }
+        self.0.max_concurrent_multipath_paths(max_concurrent);
```

Comment:
I think we should ignore the user's setting if they set it too small. if `max_concurrent < MAX_MULTIPATH_PATHS + 1` we print a warning saying we're ignoring their setting and actually set `self.0.max_concurrent_multipath_paths(MAX_MULTIPATH_PATHS + 1)`.

Otherwise it is too easy to get hard-to-debug issues, where we'll only figure things out much too late. The issues we will get from ignoring the user-provided value will be much easier to understand and then we can figure out what the usecases are and how to support them.

Same for the others.